src/storage/v2/delta_container.hpp

```diff
@@ -0,0 +1,286 @@
+// Copyright 2024 Memgraph Ltd.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt; by using this file, you agree to be bound by the terms of the Business Source
+// License, and you may not use this file except in compliance with the Business Source License.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+#pragma once
+
+#include "storage/v2/delta.hpp"
+#include "utils/allocator/page_aligned.hpp"
+#include "utils/allocator/page_slab_memory_resource.hpp"
+#include "utils/static_vector.hpp"
+
+#include <forward_list>
+
+namespace memgraph::storage {
+namespace {
+
+template <typename T>
+using PageAlignedList = std::forward_list<T, utils::PageAlignedAllocator<T>>;
+
+// assumption `sizeof(void *)` if for the node pointer inside forward_list's node
+// multiple pages is also fine, unused pages will not add to RSS
+constexpr auto kRemainingPageSpace = (4 * utils::PageAlignedAllocator<Delta>::PAGE_SIZE) - sizeof(void *);
```

Comment:
Can you put a comment that this was due to Jemalloc if I remember correctly, there was some reason you put here 4